### DISCUSSION TITLE
Data from MongoDB won't display

### DISCUSSION THREAD

**Discussion by sirtobiwan** [neuefische/questions#585] [Web] [answered]
Hi, my data from MondoDb won't display. 
I need the sum of objects from the stored array at mongoDB. 
It should display at my card component with the sum 1267519. But even with some console.logs the array is always empty.

Here you can find the stored data:
![Bildschirm­foto 2023-04-28 um 14 55 37 (2)](https://user-images.githubusercontent.com/125971942/235153410-67dcc10f-ddf4-4fe8-b457-e77fd848cd6b.png)

I created the .env, the db.model:
![Bildschirm­foto 2023-04-28 um 14 59 45](https://user-images.githubusercontent.com/125971942/235154280-0d01a4eb-2f78-449a-80f4-4e49be908cb4.png)

the api:
![Bildschirm­foto 2023-04-28 um 15 00 35](https://user-images.githubusercontent.com/125971942/235154459-9cc82b69-9cef-4d32-b133-4eb7b290aca7.png)

the map method at the card component:

![Bildschirm­foto 2023-04-28 um 15 01 31](https://user-images.githubusercontent.com/125971942/235154689-f2114308-ccf0-4659-9409-781bb556505f.png)

Here is the link to the branch: [branch_us3-fetch-data-from-mondoDB](https://github.com/sirtobiwan/Capstone-Project-Solar-Calculator/tree/us3-fetch-data-from-mondoDB)

I tried with some console logs and to google it, even ask chatgpt but can't find the issue.

Thanks for your help

**Comment by NikRoe** [comment]
Hi Tobi,

when working with useSWR, you need to provide a fetcher method. You can either do this [globally](https://swr.vercel.app/docs/global-configuration) in the _app.js or [locally](https://swr.vercel.app/docs/getting-started) in your Card component.
A simple fetcher can look like this:
```js
const fetcher = (...args) => fetch(...args).then(res => res.json())
```

and needs to be passed as a second argument to useSWR:
```js
useSWR('/api/user/123', fetcher) 
```

You should work with absolute paths instead of relative paths to fetch your API:
`'/api/ghi'`

I hope this already fixes the issue.

In case it doesn't: Please send screenshots of error messages, should there be any :)

**Comment by sirtobiwan** [comment]
Thanks Niklas.

I changed it accordingly:
![Bildschirm­foto 2023-04-28 um 15 44 35](https://user-images.githubusercontent.com/125971942/235165048-734e64e9-6459-4c26-b007-831398a8dcab.png)


Now there is a new error message 404 in the console.log:
![Bildschirm­foto 2023-04-28 um 15 44 48](https://user-images.githubusercontent.com/125971942/235164969-dc9ce664-1135-415c-95a4-9f37a66ca04d.png)

it seems that there is an issue with the api itself but I cannot find the reason for it.

**Comment by sirtobiwan** [comment, marked as answer]
issue was on side of the database.

The objects had a different key than the schema (capital letters).

Everything was sorted
